python/ccf/receipt.py

```diff
@@ -51,3 +51,14 @@ def check_endorsement(endorsee: Certificate, endorser: Certificate):
     endorser_pk.verify(
         endorsee.signature, digest, ec.ECDSA(utils.Prehashed(digest_algo))
     )
+
+
+def check_endorsements(
+    node_cert: Certificate, service_cert: Certificate, endorsements: List[Certificate]
+):
+    cert_i = node_cert
+    if endorsements:
```

Comment:
That check isn't necessary with `endorsements` not being optional (as declared), but if you want to be defensive and make it optional, you could `for endorsement in endorsements or []:`.

Reply:
Yeah, just being defensive, I added the `or []`

Reply:
Ok, fine, I think it's just a little weird, because there are two level of validation that make sense to me:

1. Assume the stuff passed in is fine, the contract is what's stated in the docstring or the type definition.
2. Be defensive, and runtime check our assumptions, in which case, the right thing to do here is check for `iterable` before we continue.

Instead, this kind of a weird in-the-middle where we say "Assume it's fine, unless it coerces to False, in which case we assume it's an empty iterable, but we don't check or handle anything else at runtime", which is weirdly specific? If `endorsements` was `Optional[List[..]]`, then sure, that's 1.